imageblock.block.inc

```diff
@@ -404,4 +412,33 @@ class BlockImage extends Block {
     $config->save();
   }
 
+  /**
+   * Try to extract a width value from an image style.
+   *
+   * Image styles do not necessarily scale or crop, so a width value may exist
+   * or not.
+   *
+   * @param array $image_style
+   *   The image style definitions.
+   *
+   * @return
+   *   Number (width in pixels) or FALSE if the style doesn't resize.
+   */
+  protected function extractEffectWidth($image_style) {
+    $style_width = &backdrop_static(__METHOD__);
```

Comment:
Reminder to self: this should probably be an array, keyed by style name. There could be multiple per page.